program/src/lib.rs

```diff
@@ -94,16 +94,21 @@
 //!
 //! ```toml
 //! [lib]
-//! crate-type = ["cdylib", "rlib"]
+//! crate-type = ["cdylib"]
 //!
 //! [features]
 //! no-entrypoint = []
 //! ```
 //!
-//! Note that a Solana program must specify its crate-type as "cdylib", and
-//! "cdylib" crates will automatically be discovered and built by the `cargo
-//! build-bpf` command. Solana programs also often have crate-type "rlib" so
-//! they can be linked to other Rust crates.
+//! Note that a Solana program must specify its crate-type as "cdylib", to
+//! be discovered and built by the `cargo-build-sbf` command as a deployable program.
+//! Solana programs also often have crate-type "rlib" so they can be linked to other Rust crates.
+//! Avoid using "rlib" and "cdylib" crates together, since their combined usage precludes
+//! compiler optimizations that may decrease program size and CU usage.
+//!
+//! Prefer writing a separate package if it is supposed to be used as a library for other Solana
+//! programs (i.e. a "rlib" only crate). When created a Rust project intended to be a program
```

Comment:
Wondering if we can expand a bit this to add something along the lines:

```rust
//! This would be normally the case for defining account types and helpers that are used by both clients and program.
```